rust/rsmgp-sys/src/vertex/mod.rs

```diff
@@ -0,0 +1,204 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ffi::{CStr, CString};
+
+use crate::edge::*;
+use crate::memgraph::*;
+use crate::mgp::*;
+use crate::property::*;
+use crate::result::*;
+use crate::value::*;
+// Required here, if not present tests linking fails.
+#[double]
+use crate::mgp::ffi;
+use mockall_double::double;
+
+pub struct VerticesIterator {
+    ptr: *mut mgp_vertices_iterator,
+    is_first: bool,
+    memgraph: Memgraph,
+}
+
+impl VerticesIterator {
+    pub fn new(ptr: *mut mgp_vertices_iterator, memgraph: &Memgraph) -> VerticesIterator {
+        VerticesIterator {
+            ptr,
+            is_first: true,
+            memgraph: memgraph.clone(),
+        }
+    }
+}
+
+impl Drop for VerticesIterator {
+    fn drop(&mut self) {
+        unsafe {
+            if !self.ptr.is_null() {
+                ffi::mgp_vertices_iterator_destroy(self.ptr);
+            }
+        }
+    }
+}
+
+impl Iterator for VerticesIterator {
+    type Item = Vertex;
+
+    fn next(&mut self) -> Option<Vertex> {
+        unsafe {
+            let data: *const mgp_vertex;
+            if self.is_first {
+                self.is_first = false;
+                data = ffi::mgp_vertices_iterator_get(self.ptr);
+            } else {
+                data = ffi::mgp_vertices_iterator_next(self.ptr);
+            }
+
+            if data.is_null() {
+                None
+            } else {
+                Some(match Vertex::mgp_copy(data, &self.memgraph) {
+                    Ok(v) => v,
+                    Err(_) => panic!("Unable to create new vertex during vertices iteration."),
+                })
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Vertex {
+    ptr: *mut mgp_vertex,
+    memgraph: Memgraph,
+}
+
+impl Drop for Vertex {
+    fn drop(&mut self) {
+        unsafe {
+            if !self.ptr.is_null() {
+                ffi::mgp_vertex_destroy(self.ptr);
+            }
+        }
+    }
+}
+
+impl Vertex {
+    pub fn new(ptr: *mut mgp_vertex, memgraph: &Memgraph) -> Vertex {
+        Vertex {
+            ptr,
+            memgraph: memgraph.clone(),
+        }
+    }
+
+    /// Creates a new Vertex based on [mgp_vertex].
+    pub(crate) unsafe fn mgp_copy(
+        mgp_vertex: *const mgp_vertex,
+        memgraph: &Memgraph,
+    ) -> MgpResult<Vertex> {
+        // Test passes null ptr because nothing else is possible.
+        #[cfg(not(test))]
+        assert!(
+            !mgp_vertex.is_null(),
+            "Unable to make vertex copy because vertex pointer is null."
+        );
```

Comment:
Why the `is_null` check is present here but not in the `new` function? I see this pattern in other structs also (edge,map,path). I think ensuring the pointer is not null would be beneficial, because the non-static functions don't do any null checks and calling c function with null pointer is dangerous.